students/Vincent_Aquila/session03/strformat_lab.py

```diff
@@ -0,0 +1,85 @@
+"""
+Python210A Vincent Aquila  Fall2018
+using Python 3.7.0
+assignment objective: String Formatting Lab
+"""
+
+
+# task 1
+print("---Task 1")
+"""
+Write a format string that will take the following four element tuple:
+( 2, 123.4567, 10000, 12345.67)
+and produce:
+'file_002 :   123.46, 1.00e+04, 1.23e+04'
+"""
+t1 = (2, 123.4567, 10000, 12345.67)
+print("file_{:03} : {:.2f} {:.2e} {:.2e}".format(t1[0], t1[1], t1[2], t1[3]))
+
+
+# task 2
+print("---Task 2")
+"""
+Using your results from Task One, repeat the exercise, but this time using an alternate
+type of format string (hint: think about alternative ways to use .format() (keywords anyone?),
+and also consider f-strings if you’ve not used them already).
+"""
+t2 = (2, 123.4567, 10000, 12345.67)
+print(f'file_{t2[0]:03} : {t2[1]:.2f} {t2[2]:.2e} {t2[3]:.2e}')
+
+
+# task 3
+print("---Task 3")
+"""
+Rewrite:
+"the 3 numbers are: {:d}, {:d}, {:d}".format(1,2,3)
+to take an arbitrary number of values.
+"""
+def display_items(seq):
+    l = len(seq)
+    print(("There are {} items, and they are: " + ", ".join(["{}"]*l)).format(l, *seq))
+display_items([2,4,7,12,5,19])
+
+
+# task 4
+print("---Task 4")
+"""
+Given a 5 element tuple:
+(4, 30, 2017, 2, 27)
+use string formating to print:
+'02 27 2017 04 30'
+Hint: use index numbers to specify positions.
+"""
+t4 = (4, 30, 2017, 2, 27)
+print("{:02} {} {} {:02} {}".format(t4[3], t4[4], t4[2], t4[0], t4[1]))
```

Comment:
we were going for putting teh indexes in the format specifier:

"{3:02}" etc....